src/main/java/duke/TasksList.java

```diff
@@ -0,0 +1,179 @@
+package duke;
+
+import duke.exception.InvalidArgumentException;
+import duke.exception.InvalidIndexException;
+import duke.task.Task;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+public class TasksList {
+    private List<Task> tasks;
+
+    /**
+     * TasksList constructor method.
+     * Uses ArrayList<Task> as its core.
+     */
+    public TasksList() {
+        this.tasks = new ArrayList<Task>();
+    }
+
+    /**
+     * Adds user input task to the taskslist.
+     *
+     * @param instructions List of user input words.
+     * @return Response text to be printed.
+     * @throws InvalidArgumentException If user input argument format is unknown/invalid.
+     */
+    public String addTask(List<String> instructions) throws InvalidArgumentException {
+        Task newTask = Task.createTask(instructions);
+        tasks.add(newTask);
+        String response;
+        response = String.format("Got it. I've added this task\n" + newTask + "\n" +
+                "You have %d tasks in the list", this.tasks.size());
+        return response;
+    }
+
+    /**
+     * Deletes task of certain index from the taskslist.
+     *
+     * @param index Index of the task (1 based).
+     * @return Response text to be printed.
+     * @throws InvalidIndexException If index is out of bounds.
+     */
+    public String deleteTask(int index) throws InvalidIndexException {
+        if (index > this.tasks.size()) {
+            throw new InvalidIndexException();
+        }
+
+        String response = String.format("Ok, I will remove this task: \n %s", tasks.get(index - 1));
+        tasks.remove(index - 1);
+        return response;
+    }
+
+    /**
+     * Returns the list of tasks in the form of String / text.
+     *
+     * @return List of tasks with numberings in text format.
+     */
+    public String list() {
```

Comment:
To increase readability, you could consider changing the method name to "getListOfTasks".